src/map.rs

```diff
@@ -44,6 +44,14 @@ impl MapDataList {
             self.maps.get_mut(map_index).map(|v| &mut v.2)
         }
     }
+
+    pub fn map_mut(&mut self, i: usize) -> Option<&mut MapData> {
+        if self.maps.is_empty() || i >= self.maps.len() {
+            None
+        } else {
+            self.maps.get_mut(i).map(|v| &mut v.2)
+        }
+    }
```

Comment:
If/else is redundant
```suggestion
    pub fn map_mut(&mut self, i: usize) -> Option<&mut MapData> {
        self.maps.get_mut(i).map(|v| &mut v.2)
    }
```